olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,229 @@
+            ),
+        }
+
+    @torch.no_grad()
+    def _run_for_config(
+        self, model: PyTorchModelHandler, data_root: str, config: Dict[str, Any], output_model_path: str
+    ) -> ONNXModelHandler:
+        from auto_gptq import BaseQuantizeConfig
+        from auto_gptq.modeling import BaseGPTQForCausalLM
+        from auto_gptq.modeling.auto import GPTQ_CAUSAL_LM_MODEL_MAP
+
+        from olive.passes.onnx.gptq_utils import QuantLinearORT
+
+        if self.accelerator_spec.accelerator_type != Device.GPU:
+            raise ValueError("Please use GPU to run gptq quantization.")
+
+        dataset = None
+        if config["dataloader_func"]:
+            dataset = self._user_module_loader.call_object(
+                config["dataloader_func"],
+                **(config["dataloader_func_kwargs"] or {}),
+            )
+        elif config["data_config"]:
+            data_config = validate_config(config["data_config"], DataConfig)
+            dataloader = data_config.to_data_container().create_dataloader(data_root)
+            dataset = [data[0] for data in dataloader]
+
+        if (
+            not dataset
+            or not isinstance(dataset, list)
+            or not isinstance(dataset[0], dict)
+            or ("input_ids" not in dataset[0] or "attention_mask" not in dataset[0])
+        ):
+            raise ValueError(
+                "Provided dataset is invalid. The returned datasets is a list of tokenized data "
+                "(e.g. [{ 'input_ids': [ 1, 100, 15, ... ],'attention_mask': [ 1, 1, 1, ... ]},...])"
+            )
+
+        pytorch_model = model.load_model()
+        quantize_config = BaseQuantizeConfig(
+            bits=config["bits"],
+            group_size=config["group_size"],
+            damp_percent=config["damp_percent"],
+            static_groups=config["static_groups"],
+            true_sequential=config["true_sequential"],
+            desc_act=config["desc_act"],
+            sym=config["sym"],
+        )
+
+        def get_onnx_quant_linear(*args, **kwargs):
+            return QuantLinearORT
+
+        if hasattr(pytorch_model, "config") and pytorch_model.config.model_type in GPTQ_CAUSAL_LM_MODEL_MAP:
+            model_type = pytorch_model.config.model_type
+            model_class = GPTQ_CAUSAL_LM_MODEL_MAP[model_type]
+            quantized_model = model_class(pytorch_model, False, quantize_config)
+        else:
+            quantized_model = BaseGPTQForCausalLM(pytorch_model, False, quantize_config)
+            if not (config["layers_block_name"] and config["outside_layer_modules"] and config["inside_layer_modules"]):
+                raise ValueError(
+                    "Can't get layers_block_name to quantize automatically, "
+                    "please set layers_block_name, outside_layer_modules and inside_layer_modules in config."
+                )
+            quantized_model.layers_block_name = config["layers_block_name"]
```

Comment:
do we need specify https://github.com/AutoGPTQ/AutoGPTQ/blob/896d8204bc89a7cfbda42bf3314e13cf4ce20b02/auto_gptq/modeling/_base.py#L201

Reply:
I searched layer_type and it is only needed in from_pretrained() and from_quantized() which we will not call.
https://github.com/AutoGPTQ/AutoGPTQ/blob/896d8204bc89a7cfbda42bf3314e13cf4ce20b02/auto_gptq/modeling/_base.py#L764-L776
https://github.com/AutoGPTQ/AutoGPTQ/blob/896d8204bc89a7cfbda42bf3314e13cf4ce20b02/auto_gptq/modeling/_base.py#L1048-L1060